build-tools/packages/version-tools/src/semver.ts

```diff
@@ -32,17 +34,36 @@ export function bumpRange(
 	bumpType: VersionBumpTypeExtended,
 	prerelease = false,
 ): string {
-	if (semver.validRange(range) === null) {
-		throw new Error(`${range} is not a valid semver range.`);
+	const [isWorkspaceProtocol, rangeToBump] = parseWorkspaceProtocol(range);
```

Comment:
`bumpRange` seems like a complicated operation that includes a lot of policy, and I don't understand why we should need or want it.

If you are updating deps on one release group to depend on an updated version  X of another release group, it seems much simpler to do something more like:

`generateVersionConstaint(useWorkspace, X, rangeTypeEnum)`

I don't see why we would want to compute these versions in terms of some existing range or use bumping. Instead of just explicitly computing the version string we want (and optionally providing it via the command line) like the new type test generation does seems so much simpler. Removing the dependency on the old version lets you compute the version string once at the beginning of the operation, and the tool can be used to cleanup/fix messed up versions already in package.json.

Reply:
Looks like this is just used by "bumpPackageDependencies".

Can our work flows instead be adjusted such that instead of "bumpPackageDependencies", we have an option to set what version of one release group another release group depends on? Ex: Set client to depend on "workspace:*" of client, or "^1.2.3" of common-utils for example? I feel like building tooling around "set" instead of "bump" makes everything much simpler (have less dependencies, recover from errors better, support changes to policy without code cahnges easier etc).

Reply:
> I don't see why we would want to compute these versions in terms of some existing range or use bumping.

I think we could factor things out much better. There are ultimately two parts - figuring out what version to set,
then setting it. "Bumping" sort of combines them. I think we mix the two in more places than we should, but we mostly
follow that general pattern within the implementations of the "bump" functions.

The version-tools CLI is used by some partners to integrate with non-JS build systems, and in those cases they just have
a package.json file with a dependency range and want to bump to the latest/next major/minor range, whatever it is.
That's the origin of the bumpRange function.

> Can our work flows instead be adjusted such that instead of "bumpPackageDependencies", we have an option to set what
> version of one release group another release group depends on?

I think what you're describing is more or less the bumpReleaseGroup function, but it mixes "figure out the version" with
"set the version" again.

Reply:
The latest refactoring should address this.